Add multi_file docs and port over more docs stuff

sebastian-quintero opened the pull request:
# Description

Adds a whole lotta documentation to the Python SDK, including how to handle multi-file. From the docs2 side, all _internal_ references to the Python SDK were removed, so any place that had Python scripts were ported here, under the correct section. Docs are still not great, but it's something, TM.

Fixes some minor bugs in functions for handling multi-file and text when uploading.

sebastian-quintero committed:
Multi-file docs

sebastian-quintero committed:
Adds tutorials for cloud application usage

Adds new tutorials covering execution classes, large payloads,
queuing/prioritization, and general application runs, with
instructions and code examples.

These tutorials aim to guide users on how to run applications, handle
large payloads, manage queuing and prioritization, and understand
execution classes, providing comprehensive information for effective
utilization of Nextmv Cloud.

Updates the Application class to resolve the format of the input
data, using the values provided by the user if they are present,
but defaulting to JSON if a dict is provided.

sebastian-quintero committed:
Adds experiment tutorials

merschformann commented on docs/nextmv/tutorials/input.md:
````diff
@@ -55,3 +67,124 @@ print(csv_archive_input_1.data)
 print(csv_archive_input_2.data)
 ```
+
+## `MULTI_FILE` inputs
````
I used the `DataFile` class with `input_data_key`, which felt necessary for a multi-sheet Excel, in my multi-excel example app [here](https://github.com/nextmv-io/community-apps/blob/merschformann/python-multi-excel/python-ortools-multiknapsack-multiexcel/main.py#L32-L43). Maybe we can add a custom example like that here as well? I actually liked working with it quite a bit. And again, I think it's necessary due to overlapping file/key for multi-sheet Excel (which are quite the norm).
Btw, I hope the example app mostly makes sense to you. It is mainly aligned with the Java one that came first, but maybe it needs another spin. :grimacing:

sebastian-quintero replied:
Thank you for your suggestion, I added an example using the `input_data_key`.

merschformann reviewed (approved):
Awesome! Thanks for all the work! :heart: 
I only had a minor comment. :blush:

merschformann commented on docs/nextmv/tutorials/input.md:
```diff
@@ -4,11 +4,21 @@
+holding place for a decision model's input data. An input is built through
+[options][options] and data. An input is loaded from a source, through the
+[`InputLoader`][input-loader] class. You may use the [`load`][load] function to
+build an input from a source, or call the `.load`method on the `InputLoader`
```
Yeah, fair one.

sebastian-quintero replied:
Good little bot.

merschformann commented on nextmv/nextmv/cloud/run.py:
```diff
@@ -470,6 +470,41 @@ class RunConfiguration(BaseModel):
+        if self.format is not None:
+            return
+
+        if self.format is None:
```
Yeah, I think we can just remove the `if` check here (not the body).

sebastian-quintero replied:
Good little robot you, who is a good little robot?

sebastian-quintero commented on nextmv/nextmv/cloud/application.py:
```diff
@@ -1619,10 +1619,8 @@ def new_run(  # noqa: C901 # Refactor this function at some point.
         upload_id_used = upload_id is not None
-
-        if not upload_id_used and upload_url_required:
+        if self.__upload_url_required(upload_id_used, input_size, tar_file, input):
```
AI stop hallucinating, of course the idea is that it returns true, because it means that I _must_ use an upload URL.

sebastian-quintero committed:
Adds input data key customization example

Extends the input tutorial with an example that
demonstrates how to customize the key name for data
loaded from various file types.

Simplifies conditional logic for determining input format in
RunConfiguration.
